mailpoet/lib/Segments/SegmentsRepository.php

```diff
@@ -136,6 +143,28 @@ public function verifyNameIsUnique(string $name, ?int $id): void {
     }
   }
 
+  /**
+   * @param int $id
+   *
+   * @return SegmentEntity
+   * @throws InvalidStateException
+   */
+  public function verifyFilterSegmentExists(int $id): SegmentEntity {
```

Comment:
I wonder if we should change the name of this method to indicate that it also checks if the segment is dynamic since it throws an exception if another type of segment is passed?